ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/-components/packages-statistics-card.tsx

```diff
@@ -100,11 +100,16 @@ export const PackagesStatisticsCard = ({
         <ListTree className={`h-4 w-4 ${getStatusFontColor(status)}`} />
       )}
       value={status ? packagesTotal : 'N/A'}
-      description={
-        ecoSystems.length > 1
-          ? `from ${ecoSystems.length} ecosystems (${ecoSystems.join(`,`)})`
-          : `from 1 ecosystem (${ecoSystems})`
-      }
+      description={(() => {
+        switch (ecoSystems.length) {
+          case 0:
+            return '';
+          case 1:
+            return `from 1 ecosystem (${ecoSystems})`;
+          default:
+            return `from ${ecoSystems.length} ecosystems (${ecoSystems.join(`,`)})`;
+        }
+      })()}
```

Comment:
This would be a little more idiomatic without the function definition and call as:

```typescript
ecoSystems.length
  ? ecoSystems.length > 1
    ? `from ${ecoSystems.length} ecosystems (${ecoSystems.join(`,`)})`
    : `from 1 ecosystem (${ecoSystems})`
  : ''"
```

The number [0 is Falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), so the ternary will handle others in the truthy branch and it in the falsy.

Reply:
Another option could be to change the `description` prop of `StatisticsCard` to be `ReactNode`, in which case a function could be passed to it as is without the need of `()()`.

Reply:
Yeah, I was reading about the nested ternary operator approach over [here](https://dev.to/gvergnaud/bringing-pattern-matching-to-typescript-introducing-ts-pattern-v3-0-o1k), but deliberately decided against it (due to the hard to read nested). However, as you say it's more idiomatic, and it's fewer lines as well, I'll change it.

Reply:
Imo the most non-idiomatic part is the immediate call of the arrow function as `(() => {})**()**`, but that should be fixed by changing `description` to [`ReactNode`](https://reactnative.dev/docs/react-node), which would allow just passing the function without calling it here. As `string` is also `ReactNode`, the change should not require changes elsewhere where the component is used.

Reply:
I sticked to your code as it's more compact.